### PR TITLE
Patelchandni/ziptoperror

### PR DESCRIPTION
**Pull request by patelchandni**
Only for function app with run from package setup, if we get "Offset to Central Directory cannot be held in an Int64" error, then adding a retry to download zip file again and extract triggers.

### CODE REVIEW

**Comment by suwatch** [comment]
please rebase and squash to one commit once ready for review.

**Comment by patelchandni** [comment]
Adding @eliaslopezgt for review as well.

File: Kudu.Services/ByteRanges/HttpRequestMessageExtensions.cs [hunk 1]
```diff
@@ -41,34 +42,48 @@ public static HttpResponseMessage CreateErrorResponse(this HttpRequestMessage re
             return rangeNotSatisfiableResponse;
         }
 
-        public static async Task<long> CopyToAsync(this HttpContent content, string filePath, ITracer tracer)
+        public static async Task<long> CopyToAsync(this HttpContent content, string filePath, ITracer tracer, bool leaveOpen = false)
```
Comment (eliaslopezgt) [inline Kudu.Services/ByteRanges/HttpRequestMessageExtensions.cs:45]: Consider creating a new method, and based on a feature flag switch between them.

File: Kudu.Services/Deployment/PushDeploymentController.cs [hunk 2]
```diff
@@ -384,161 +384,172 @@ private async Task<HttpResponseMessage> PushDeployAsync(ArtifactDeploymentInfo d
...
 
-            return response;
+                return response;
+            }
+            finally
+            {
+                if (isRequestJSON != true && artifactType == ArtifactType.Zip && _settings.RunFromLocalZip()) 
+                {
+                    var src = await content.ReadAsStreamAsync();
+                    src.Dispose();
```
Comment (eliaslopezgt) [inline Kudu.Services/Deployment/PushDeploymentController.cs:550]: null check?

**Review by eliaslopezgt — CHANGES_REQUESTED** [review]
👍 to test this with some comments.

File: Kudu.Services/Deployment/PushDeploymentController.cs (hunk 2 above)
Comment (patelchandni) [reply Kudu.Services/Deployment/PushDeploymentController.cs]: added check

File: Kudu.Services/ByteRanges/HttpRequestMessageExtensions.cs (hunk 1 above)
Comment (patelchandni) [reply Kudu.Services/ByteRanges/HttpRequestMessageExtensions.cs:75]: added new method that is only called for function app with run from package path

**Pull request merged**